sycl/source/backend.cpp

```diff
@@ -0,0 +1,101 @@
+//==------------------- backend.cpp ----------------------------------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "detail/context_impl.hpp"
+#include "detail/event_impl.hpp"
+#include "detail/platform_impl.hpp"
+#include "detail/plugin.hpp"
+#include "detail/queue_impl.hpp"
+#include <CL/sycl/backend.hpp>
+#include <CL/sycl/detail/export.hpp>
+#include <CL/sycl/detail/pi.h>
+#include <CL/sycl/exception_list.hpp>
+
+#include <stdexcept>
+
+__SYCL_INLINE_NAMESPACE(cl) {
+namespace sycl {
+namespace detail {
+
+static const plugin &getPlugin(backend Backend) {
+  switch (Backend) {
+  case backend::opencl:
+    return pi::getPlugin<backend::opencl>();
+  case backend::level_zero:
+    return pi::getPlugin<backend::level_zero>();
+  default:
+    throw std::runtime_error{"Unsupported backend"};
+  }
+}
+
+__SYCL_EXPORT platform make_platform(pi_native_handle NativeHandle,
```

Comment:
```suggestion
platform make_platform(pi_native_handle NativeHandle,
```
Looks like SYCL_EXPORT is unnecessary here since the function declaration is already marked with one.